### DISCUSSION TITLE
在使用`编辑state`功能的时候产生了一点想法

### DISCUSSION THREAD

**Discussion by OneGIl** [vuejs/devtools#147] [General]
借 #144 这个pr想说一下我的一点想法（我也在尝试解决相同的问题），当前 `InspectorStateField.vue` 中，在submit时，payload中的state.value不管state是什么类型，发送的都是字符串类型，这导致修改非字符串类型的state会出现非主观的类型转换问题。比如在该pr中的视频里，修改count时，一旦在输入框输入任意“数字”，submit后会变成字符串类型，action中的+-号消失是证明。还有在component中修改ref类型的object，submit后也会变成字符串而非对象。还有一些特殊的类型（如：undefined、null、NaN、true等）也会有一些奇怪的问题。

而改进方案是参考老版本的devtool那样，比如字符串类型的value使用双引号包起来，undefined等特殊类型单独处理，其他类型做好表单验证即可。

工作量应该不小（？），但如果可以的话，我想尝试改造一下~

**Comment by webfansplz** [comment]
当然，欢迎 PR～
